From 79aa1c20640f26fa4faa608646b16e213e24228c Mon Sep 17 00:00:00 2001
From: Yuxi Sun <b36102@freescale.com>
Date: Mon, 21 Nov 2011 13:29:55 +0800
Subject: [PATCH 244/505] ENGR00160799 fix display deranged image when run
 overlay first time

Fix display deranged image when run overlay first time. Move
VIDIOC_S_OUTPUT ioctrl command  before overlay_setup function.

Signed-off-by: Yuxi Sun <b36102@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_overlay.c | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_overlay.c b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
index af4b690..1d583a9 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_overlay.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_overlay.c
@@ -612,20 +612,7 @@ main(int argc, char **argv)
 	}
 	printf("sensor chip is %s\n", chip.match.name);
 
-        fmt.type = V4L2_BUF_TYPE_VIDEO_OVERLAY;
-        fmt.fmt.win.w.top=  g_display_top ;
-        fmt.fmt.win.w.left= g_display_left;
-        fmt.fmt.win.w.width=g_display_width;
-        fmt.fmt.win.w.height=g_display_height;
-
-        if (mxc_v4l_overlay_setup(&fmt) < 0) {
-                printf("Setup overlay failed.\n");
-                return TFAIL;
-		}
-
-        memset(&fb_v4l2, 0, sizeof(fb_v4l2));
-
-        if ((fd_fb_0 = open(fb_device_0, O_RDWR )) < 0)	{
+	if ((fd_fb_0 = open(fb_device_0, O_RDWR )) < 0)	{
 		printf("Unable to open frame buffer 0\n");
                 return TFAIL;
         }
@@ -651,6 +638,19 @@ main(int argc, char **argv)
         }
 
 
+        fmt.type = V4L2_BUF_TYPE_VIDEO_OVERLAY;
+        fmt.fmt.win.w.top=  g_display_top ;
+        fmt.fmt.win.w.left= g_display_left;
+        fmt.fmt.win.w.width=g_display_width;
+        fmt.fmt.win.w.height=g_display_height;
+
+        if (mxc_v4l_overlay_setup(&fmt) < 0) {
+                printf("Setup overlay failed.\n");
+                return TFAIL;
+		}
+
+        memset(&fb_v4l2, 0, sizeof(fb_v4l2));
+
 	fb_device_fg = "/dev/fb1";
 	if ((g_fd_fb_fg = open(fb_device_fg, O_RDWR)) < 0) {
 		printf("Unable to open frame buffer 1\n");
-- 
1.8.0

